src/chain/chain.rs

```diff
@@ -354,7 +357,7 @@ impl Chain {
             _ => false,
         };
 
-        if start_churn_event && (self.churn_in_progress || self.relocation_in_progress) {
+        if start_churn_event && !self.can_poll_churn() {
```

Comment:
just to confirm that the `can_poll_churn` function will check two additional flag `handled_genesis_event` and `split_in_progress` from state.
Is this ok?

Reply:
Yes, it should be the same condition that allow an Online/Offline event to be coming out of that function